app/controllers/api/v1/products_controller.rb

```diff
@@ -136,9 +138,11 @@ def destroy
     def set_product
       scoped_products = authorized_scope(current_account.products)
 
-      @product = scoped_products.find(params[:id])
-
-      Current.resource = product
+      Current.resource = @product = FindByAliasService.call(
```

Comment:
I'd like to eventually move this into a `find_by_alias` concern, where a model can define aliases:

```rb
class Product
  include Aliasable
  
  has_aliases :code
end
```